build-support/bin/cherry_pick.sh

```diff
@@ -25,16 +25,33 @@ if [[ -z $PR_NUM ]]; then
   read -r PR_NUM
 fi
 
-MILESTONE=$2
-if [[ -z $MILESTONE ]]; then
-  MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
-  if [[ -z $MILESTONE ]]; then
+TARGET_MILESTONE=$2
+if [[ -z $TARGET_MILESTONE ]]; then
+  TARGET_MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
+  if [[ -z $TARGET_MILESTONE ]]; then
     echo "No milestone on PR. What's the milestone? (E.g. 2.10.x)"
-    read -r MILESTONE
+    read -r TARGET_MILESTONE
   fi
 fi
 
+# NB: Find all milestones >= $TARGET_MILESTONE by having GH list them, then uses awk to trim the
+# results to $TARGET_MILESTONE and all milestones after it. Later we will verify these milestones
+# match a branch name.
+# shellcheck disable=SC2016
+MILESTONES=$(gh api graphql -F owner=":owner" -F name=":repo" -f query='
+  query ListMilestones($name: String!, $owner: String!) {
+    repository(owner: $owner, name: $name) {
+      milestones(last: 10) {
+        nodes {title}
+      }
+    }
+  }' --jq .data.repository.milestones.nodes.[].title | awk "/$TARGET_MILESTONE/{p=1}p" -)
```

Comment:
So here's what's unclear: Why are we using $TARGET_MILESTONE as a regex? It is intended to be a literal string. And why are we searching for it as the pattern? And what does the action do? That apparent magic is what I think needs documenting.

Reply:
Ah ok that's much more explicit if a concern

Reply:
From my testing, this appears to rely on the fact that the milestones are returned in increasing order, so that must be verified as a property of the API (or if not, we must sort), and must be documented.

Reply:
They're in that order because we create them in that order. I can't imagine we'd be making milestones for later branches before earlier ones

Reply:
And just generally explaining the awk program would be really helpful.

Reply:
And yet if we do, this will fail in a very unpredictable and undocumented way. And `sort` is free, as is documenting why.

Reply:
`sort` isn't correct though, right? 

```
$ gh api graphql -F owner=":owner" -F name=":repo" -f query='
>   query ListMilestones($name: String!, $owner: String!) {
>     repository(owner: $owner, name: $name) {
>       milestones(last: 10) {
>         nodes {title}
>       }
>     }
>   }' --jq .data.repository.milestones.nodes.[].title | sort
2.10.x
2.11.x
2.12.x
2.13.x
2.14.x
2.15.x
2.16.x
2.17.x
2.8.x
2.9.x
```

Reply:
Oh hot damn there's a version sort?!?!?!?!

`  -V, --version-sort          natural sort of (version) numbers within text`